src/de.rs

```diff
@@ -406,9 +406,19 @@ struct Enum<'de> {
 
 impl<'de> Enum<'de> {
     fn new(ev: &'de [Document]) -> Result<Self> {
-        let (e, v) = match ev.len() {
+        // Count up value-containing document nodes. We expect only one and need
+        // to ignore any comment-nodes if present.
+        let mut values = 0;
+        let mut node = 0;
+        for (i, val) in ev.iter().enumerate() {
+            if val.has_value() {
+                values += 1;
+                node = i;
+            }
+        }
+        let (e, v) = match values {
             0 => Err(Error::StructureError("one value", "none")),
-            1 => ev[0].as_kv(),
+            1 => ev[node].as_kv(),
             _ => Err(Error::StructureError("one value", "many")),
         }?;
         Ok(Enum { enm: e, var: v })
```

Comment:
I prefer using a filtered iterator, something like this (I haven't verified this compiles):

```rust
// We expect exactly one of the document nodes will contain a value
// (as opposed to a comment). Extract that node.
let mut with_values = ev.iter().map(Document::has_value);
let ev = with_values.next().ok_or_else(|| Error::StructureError("one value", "none"))?;
if with_values.next().is_some() {
    return Err(Error::StructureError("one value", "many"));
}
let (e, v) = ev.as_kv()?;
Ok(Enum { enm: e, var: v })
```

Purely subjective though, so I'm leaving the style up to you.

Reply:
I like the iterator form better.  Updated.